server/routes.go

```diff
@@ -1529,6 +1530,24 @@ func (s *Server) ChatHandler(c *gin.Context) {
 		defer close(ch)
 		var sb strings.Builder
 		var toolCallIndex int = 0
+		var templateToolToken string
+		var tmpl *gotmpl.Template
+		if len(req.Tools) > 0 {
```

Comment:
Shouldn't this error on "model does not support tools" vs getting to this point?

Reply:
Nope. We support models with a tool token.

Reply:
Ah I see. Do we need this code block then? Some models may not have a prefix ahead of the JSON or python tool call by design and this logging statement makes it sound like an issue. Plus, hopefully that case is rare! (e.g. Gemma3 with python still has a leading `[` I believe, which we can look for)